lib/Doctrine/DBAL/Connection.php

```diff
@@ -402,7 +403,7 @@ private function detectDatabasePlatform()
     {
         $version = $this->getDatabasePlatformVersion();
 
-        if (null !== $version) {
+        if ($version !== null && $this->_driver instanceof VersionAwarePlatformDriver) {
```

Comment:
Wouldn't it be enough to check if the driver is not null? Type hint should help static analysis w/o the run time check.

Reply:
`$this->_platform` is typed as AbstractPlatform, but `createDatabasePlatformForVersion()` is only defined in some drivers implementing VersionAwarePlatformDriver.
I at least replaced this by `assert()` inside the `if` since `Connection::getDatabasePlatformVersion()` returns null for non-version-aware drivers.